components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2092,14 +2097,27 @@ private void validateAndSetAPISecurity(API api) {
         if (log.isDebugEnabled()) {
             log.debug("API " + api.getId() + " has following enabled protocols : " + apiSecurity);
         }
-        api.setApiSecurity(apiSecurity);
+
+        String apiType;
+        if (apiProduct == null) {
+            apiType = "API";
+            api.setApiSecurity(apiSecurity);
+        } else {
+            apiType = "APIProduct";
+            apiProduct.setApiSecurity(apiSecurity);
+        }
```

Comment:
Please use the constant that is already defined to set the apiType for consistency

Reply:
Fixed in 32e0ed7cffd83e1e1bfb9eaf3eed84e0bb15a83a